### DISCUSSION TITLE
Using VScode with ssh remote server through jump servers

### DISCUSSION THREAD

**Discussion by akrz91** [blinksh/blink#1516] [Support] [answered]
Hello,

I’m not raising an issue yet as I don’t know if what I observe is an expected behavior or some feature missing.

I’m using blink shell to connect to the remote server and it is properly handled in Files app as a Location. Thanks to that I can open a VScode for WEB with a remote filesystem attached, wow! :)

But the problems that I observe start to appear when the server is not a direct connection, but between iPad and my server there is any other “jump server”.
In my case there are two actually but it does not matter as in a single jump server case I observe the same.

When connecting to ssh via jump server I have no problem in command line but the location on Files App reports “unknown error” and claims it cannot show the folder content. When running `code <host>:<path>` command it is also only launching the VScode instance but no remote files are visible.

In logs I can only see some generic error:
`[error] Jul 23 2022, 20:17:18 Connection ::1.49870 Error completing operation - connError(msg: "Socket error: disconnected")`

I’ve read here: https://docs.blink.sh/advanced/files-app
that proxyJump should be supported for the Files app but it seem like it is not working for me.

Is there anything I can do with it?
Have anyone experienced similar issue?

Thanks and regards,
Adam

**Comment by carloscabanero** [comment, marked as answer]
Hey Adam, it looks like possibly an issue with the configuration. Could you share how you are setting things up?

There is more information https://docs.blink.sh/advanced/advanced-ssh#jumpbastion-hosts. Another thing to try is to connect from the CLI first, make sure that works. And also, make sure that the destination has enabled SFTP.

**Reply by akrz91** [reply]
Hello Carlos,

Thanks for your answer.
It seems like I was following the same instruction as you provided but something had to be wrong in my configuration.
Now when I configured the jump servers once again it works fine :)

Thanks!
